src/lib/utils/SynapseClient.ts

```diff
@@ -112,14 +112,20 @@ const fetchWithExponentialTimeout = <T>(
         return retryFetch<T>(url, options, delayMs)
       }
       return resp
-        .json()
-        .then(json => {
-          return resp.ok ? json : Promise.reject<T>(json)
+        .text()
+        .then(text => {
+          // try to parse it as json
+          try {
+            const json = JSON.parse(text)
+            return resp.ok ? Promise.resolve(json) : Promise.reject<T>(json)
+          } catch (error) {
+            // failed to parse json, return text
+            return resp.ok ? Promise.resolve(text) : Promise.reject<T>(text)
+          }
         })
         .catch((error: SynapseError) => {
           if (resp.ok) {
```

Comment:
this block (if lines 127-130) can be removed because in the then above it will always handle when the response is okay